### PR TITLE
test: convert test__init__.py from unit test to pytest

### PR DESCRIPTION
**Pull request by csemanish12**
Refactored tests/unittests/test__init__.py to use pytest instead of unittest.TestCase as part of the pytest migration effort.
  - Removed TestCase inheritance
  - Replaced setup method with fixtures
  - Maintained all original test functionality

  Related: canonical#6427

### CODE REVIEW

**Comment by csemanish12** [comment]
@blackboxsw can you please review this PR.

File: tests/unittests/test__init__.py [hunk 1]
```diff
@@ -148,76 +161,78 @@ def test_normal_version_2(self):
...
             settings.PER_INSTANCE,
         )
 
-    def test_modfreq_per_always(self):
+    def test_modfreq_per_always(self, handler_data):
         """
         C{handle_part} is called regardless of frequency if nofreq is always.
         """
-        self.frequency = "once"
+        handler_data.frequency = "once"
```
Comment (blackboxsw) [inline tests/unittests/test__init__.py]: Since we are mutating the handler_data.frequency fixture in this unittest, we may need some safeguards to ensure parallel unittest runs don't trample on each other by mutating data which is shared amongst test runs.  I recognize pytest fixtures default to function-level which I believe means we will create safe, mutable handler_data instances, but if that scope were to change this future, our tests could get exposed to intermittent failures based on order of execution failures if run in parallel as some of the mutated frequency settings could be leaked to other tests.  To avoid this potential corner case, let's make the handler_data a factory function which we will initialize inside the specific unittests

```diff
diff --git a/tests/unittests/test__init__.py b/tests/unittests/test__init__.py
index 4d5a58f19..43e38fd3d 100644
--- a/tests/unittests/test__init__.py
+++ b/tests/unittests/test__init__.py
@@ -46,15 +46,18 @@ def m_write_file():
 
 
 @pytest.fixture
-def handler_data():
-    return SimpleNamespace(
-        data="fake data",
-        ctype="fake ctype",
-        filename="fake filename",
-        payload="fake payload",
-        frequency=settings.PER_INSTANCE,
-        headers={"Content-Type": "fake ctype"},
-    )
+def handler_data_factory():
+    def _make_handler_data(frequency=settings.PER_INSTANCE):
+        return SimpleNamespace(
+            data="fake data",
+            ctype="fake ctype",
+            filename="fake filename",
+            payload="fake payload",
+            frequency=settings.PER_INSTANCE,
+            headers={"Content-Type": "fake ctype"},
+        )
+
+    return _make_handler_data
 
 
 class FakeModule(handlers.Handler):
@@ -126,11 +129,12 @@ class TestWalkerHandleHandler:
 
 
 class TestHandlerHandlePart:
-    def test_normal_version_1(self, handler_data):
+    def test_normal_version_1(self, handler_data_factory):
         """
         C{handle_part} is called without C{frequency} for
         C{handler_version} == 1.
         """
+        handler_data = handler_data_factory()
         mod_mock = mock.Mock(
             frequency=settings.PER_INSTANCE, handler_version=1
         )
@@ -151,11 +155,12 @@ class TestHandlerHandlePart:
             handler_data.payload,
         )
 
-    def test_normal_version_2(self, handler_data):
+    def test_normal_version_2(self, handler_data_factory):
         """
         C{handle_part} is called with C{frequency} for
         C{handler_version} == 2.
         """
+        handler_data = handler_data_factory()
         mod_mock = mock.Mock(
             frequency=settings.PER_INSTANCE, handler_version=2
         )
@@ -177,11 +182,11 @@ class TestHandlerHandlePart:
             settings.PER_INSTANCE,
         )
 
-    def test_modfreq_per_always(self, handler_data):
+    def test_modfreq_per_always(self, handler_data_factory):
         """
         C{handle_part} is called regardless of frequency if nofreq is always.
         """
-        handler_data.frequency = "once"
+        handler_data = handler_data_factory(frequency=settings.PER_ONCE)
         mod_mock = mock.Mock(frequency=settings.PER_ALWAYS, handler_version=1)
         handlers.run_part(
             mod_mock,
@@ -200,9 +205,9 @@ class TestHandlerHandlePart:
             handler_data.payload,
         )
 
-    def test_no_handle_when_modfreq_once(self, handler_data):
+    def test_no_handle_when_modfreq_once(self, handler_data_factory):
         """C{handle_part} is not called if frequency is once."""
-        handler_data.frequency = "once"
+        handler_data = handler_data_factory(frequency=settings.PER_ONCE)
         mod_mock = mock.Mock(frequency=settings.PER_ONCE)
         handlers.run_part(
             mod_mock,
@@ -214,8 +219,9 @@ class TestHandlerHandlePart:
         )
         assert 0 == mod_mock.handle_part.call_count
 
-    def test_exception_is_caught(self, handler_data):
+    def test_exception_is_caught(self, handler_data_factory):
         """Exceptions within C{handle_part} are caught and logged."""
+        handler_data = handler_data_factory()
         mod_mock = mock.Mock(
             frequency=settings.PER_INSTANCE, handler_version=1
         )
```

Comment (csemanish12) [reply tests/unittests/test__init__.py]: Thanks for the explanation. It makes sense.
